sdk/identity/Azure.Identity/src/AzurePowerShellCredentialOptions.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using System.Runtime.InteropServices;
+using Azure.Core;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// Options for configuring the <see cref="AzurePowerShellCredential"/>.
+    /// </summary>
+    public class AzurePowerShellCredentialOptions : TokenCredentialOptions
+    {
+        private bool _usePowerShell;
+
+        /// <summary>
+        /// Set this if you want to use PowerShell (version 5 or lower) for getting the token
+        /// instead of PowerShell Core (version 6 or higher) which is the version used by default.
+        /// This can be set to true only on Windows OS.
+        /// </summary>
+        public bool UsePowerShell
+        {
+            get
+            {
+                return _usePowerShell;
+            }
+
+            set
+            {
+                // Set this to true only for Windows OS
+                _usePowerShell = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && value;
```

Comment:
It's a bit unintuitive that setting this property silently has no effect depending on platform.

Reply:
what do you advice to do? Throw an exception when setting this property on other OS than Windows?